kernel/sched.c

```diff
@@ -975,7 +975,10 @@ s32_t z_impl_k_sleep(int ms)
 {
 	s32_t ticks;
 
-	ticks = z_ms_to_ticks(ms);
+	__ASSERT((ms >= 0) || (ms == K_FOREVER),
+		 "Only positive values are accepted.");
```

Comment:
Non-negative, not positive.

Reply:
Actually, the asserts should be followed by:
```
if (ms < 0) {
   return 0;
}
```
in both cases, so that when asserts are disabled the system will behave closer to what it did before #19591 got shoveled in.

Reply:
yes, but this is protection against misuse of documented API. Probably there is no such thing in zephyr yet but what about adding such conditional ifdef with expiry date, only for next release. Something like:
```
#if ZEPHYR_VERSION <= 2.1
if (ms < 0) {
    return 0;
}
#else
   #warning "Remove this safety net"
#endif
```

Reply:
No, I won't go for that.  That Zephyr failed to clearly document an API does not justify changes that unnecessarily break existing behavior when it's trivial to preserve it.

Reply:
on the other hand, in x releases from now this will become redundant and might bring questions why this check is there. On the other hand we will switch eventually to timeout opaque struct and that check will become irrelevant and probably removed. I will add check without version guard.